python/GafferSceneUI/SceneViewUI.py

```diff
@@ -1073,13 +1076,105 @@ def __appendClippingPlaneMenuItems( menuDefinition, prefix, view, parentWidget )
 			}
 		)
 
+def __snapshotDescription( view ) :
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	if sceneGadget.getRenderer() == "OpenGL" :
+		return "Viewport snapshots are only available for rendered (non-OpenGL) previews."
+
+	return "Snapshot viewport and send to catalogue."
+
+def __snapshotToCatalogue( catalogue, view ) :
+
+	timeStamp = str( datetime.datetime.now() )
+
+	fileName = pathlib.Path(
+		view["in"].node().getContext().substitute( catalogue["directory"].getValue() )
```

Comment:
Took me a while to figure out what was happening here, but `view["in"].node()` is just `view`, right? I think I'd be tempted to use `scriptRoot.context()` instead - there's a fair chance that these individual context accessors on UI components will disappear over the long term.

Reply:
Good to know about trending away from UI component contexts, and a big "duh" to myself regarding `view["in"].node()` being `view`. Fixed in https://github.com/GafferHQ/gaffer/pull/5533/commits/792babe29290021ed8936aafab31039f06386bfe